sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -1244,6 +1269,7 @@ nonReserved
     | TRANSFORM
     | TRUE
     | TRUNCATE
+    | TYPE
```

Comment:
we should also add it to `ansiNonReserved`.

BTW is this a non-reserved keyword? @maropu

Reply:
This is not an [ANSI reserved keyword](http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36271.1550/html/blocks/BEIJHIFE.htm), but it is reserved in [MySQL](https://dev.mysql.com/doc/refman/8.0/en/keywords.html), [PostgreSQL](https://www.postgresql.org/docs/11/sql-keywords-appendix.html), and [DB2](https://www.ibm.com/support/knowledgecenter/en/SSEPEK_11.0.0/sqlref/src/tpc/db2z_reservedwords.html). It is also on the list of [potential ANSI reserved keywords](http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36271.1550/html/blocks/BEIJCJHC.htm).